src/components/select/select.spec.js

```diff
@@ -30,207 +30,256 @@ describe('<md-select>', function() {
-  }));
+    it('should remove the input-container focus state', inject(function($rootScope, $timeout) {
+      $rootScope.val = 0;
+      var element = setupSelect('ng-model="val"', [1, 2, 3]);
+      var select = element.find('md-select');
+      var controller = element.controller('mdInputContainer');
+      $timeout.flush();
+      controller.setHasValue(true);
 
-  it('should remove the input-container focus state', inject(function($rootScope, $timeout) {
-    $rootScope.val = 0;
-    var element = setupSelect('ng-model="val"', [1, 2, 3]);
-    var select = element.find('md-select');
-    var controller = element.controller('mdInputContainer');
-    $timeout.flush();
-    controller.setHasValue(true);
+      select.triggerHandler('focus');
 
-    select.triggerHandler('focus');
+      expect(element.hasClass('md-input-focused')).toBe(true);
 
-    expect(element.hasClass('md-input-focused')).toBe(true);
+      select.triggerHandler('blur');
 
-    select.triggerHandler('blur');
+      expect(element.hasClass('md-input-focused')).toBe(false);
 
-    expect(element.hasClass('md-input-focused')).toBe(false);
+    }));
 
-  }));
+    it('should remove the tabindex from a disabled element', inject(function($document) {
+      var select = setupSelect('ng-model="val" disabled tabindex="1"').find('md-select');
+      expect(select.attr('tabindex')).toBeUndefined();
+    }));
 
-  it('should remove the tabindex from a disabled element', inject(function($document) {
-    var select = setupSelect('ng-model="val" disabled tabindex="1"').find('md-select');
-    expect(select.attr('tabindex')).toBeUndefined();
-  }));
+    it('auto-infers a value when none specified', inject(function($rootScope) {
+        $rootScope.name = "Hannah";
+        var el = setupSelect('ng-model="name"', '<md-option>Tom</md-option>' +
+              '<md-option>Hannah</md-option>');
+        expect(selectedOptions(el).length).toBe(1);
+    }));
+
+    it('errors for duplicate md-options, non-dynamic value', inject(function() {
+      expect(function() {
+        setupSelect('ng-model="$root.model"', ['a', 'a']);
+      }).toThrow();
+    }));
+
+    it('errors for duplicate md-options, ng-value', inject(function($rootScope) {
+      setupSelect('ng-model="$root.model"', '<md-option ng-value="foo">Hello</md-option>' +
+            '<md-option ng-value="bar">Goodbye</md-option>');
+      $rootScope.$apply('foo = "a"');
+      expect(function() {
+        $rootScope.$apply('bar = "a"');
+      }).toThrow();
+    }));
+
+    it('watches the collection for changes', inject(function($rootScope) {
+      $rootScope.val = 1;
+      var select = setupSelect('ng-model="val"', [1, 2, 3]).find('md-select');
+      var label = select.find('md-select-value')[0];
+      expect(label.textContent).toBe('1');
+      $rootScope.val = 4;
+      $rootScope.$$values = [4, 5, 6];
+      $rootScope.$digest();
+      expect(label.textContent).toBe('4');
+    }));
+
```

Comment:
Below (like 268) is the added test; everything else already existed; just moved it around a bit for clarity.